mm2src/mm2_bitcoin/chain/src/block_header.rs

```diff
@@ -40,6 +40,7 @@ impl Serializable for BlockHeaderBits {
 const AUX_POW_VERSION_DOGE: u32 = 6422788;
 const AUX_POW_VERSION_SYS: u32 = 537919744;
 const MTP_POW_VERSION: u32 = 0x20001000u32;
+const QTUM_BLOCK_HEADER_VERSION: u32 = 536870912;
```

Comment:
Unfortunately, QTUM doesn't use a unique block header version. The 536870912 is 0x20000000 and there are BTC blocks with this version, e.g. https://www.blockchain.com/btc/block/681654 and bitcoin blocks don't have these `hash_state_root` and other fields. I also wanted to avoid adding CoinVariant but it doesn't seem possible. Most likely there will be more of such conflicting serialization implementations - not only of block headers but transactions too.

Reply:
I see, thanks!